Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -351,6 +351,35 @@ - (void)resume
     [self.sessionReplay resume];
 }
 
+- (void)start
+{
+    if (self.sessionReplay != nil) {
+        if (self.sessionReplay.isFullSession == NO) {
+            [self.sessionReplay captureReplay];
+        }
+        return;
+    }
+
+    _startedAsFullSession = YES;
+    if (SentryDependencyContainer.sharedInstance.application.windows.count > 0) {
+        // If a window its already available start replay right away
+        [self startWithOptions:_replayOptions fullSession:YES];
+    } else {
+        // Wait for a scene to be available to started the replay
+        if (@available(iOS 13.0, tvOS 13.0, *)) {
+            [_notificationCenter addObserver:self
+                                    selector:@selector(newSceneActivate)
+                                        name:UISceneDidActivateNotification];
```

Comment:
`m`: I don't see that we remove `self` from `UISceneDidActivateNotification`. We should remove observers with the most detail possible; see https://developer.apple.com/documentation/foundation/nsnotificationcenter/1413994-removeobserver

> When removing an observer, remove it with the most specific detail possible. For example, if you used a name and object to register the observer, use [removeObserver:name:object:](https://developer.apple.com/documentation/foundation/nsnotificationcenter/1407263-removeobserver) with the name and object.

Reply:
We remove it at `newSceneActivate`

Reply:
Sorry, I had a typo in the message above. Yes, we remove it, but instead we should remove observers with the most detail possible ⬆️

Reply:
Fixed